src/go/rpk/pkg/cli/topic/consume.go

```diff
@@ -221,6 +221,9 @@ func (c *consumer) consume(ctx context.Context) {
 						r.Key = decKey
 					}
 				}
+				if r.Value == nil {
+					r.Value = []byte("null")
```

Comment:
this needs to be `r.Value = nil`. As is, this is literally the string "null" being produced